### ISSUE TITLE
Multiple go declarations in a gradle build don't work anymore

### ISSUE THREAD

**Issue by dawi** [gogradle/gogradle#182]
With gogradle 0.6.5 it was possible to build multiple binaries like this:

```
build {

    targetPlatform = 'linux-amd64, darwin-amd64'

    outputLocation = 'build/${GOOS}/${GOARCH}/${PROJECT_NAME}${GOEXE}'

    doLast {
        go 'build -o build/${GOOS}/${GOARCH}/app1a${GOEXE}  gitrepo/apps/app1/app1a'
        go 'build -o build/${GOOS}/${GOARCH}/app1b${GOEXE}  gitrepo/apps/app1/app1b'
    }
}
```

Unfortunately this doesn't work anymore, because the go command needs to be configured outside the `doLast` block like this:

```
build {

    targetPlatform = 'linux-amd64, darwin-amd64'

    outputLocation = 'build/${GOOS}/${GOARCH}/${PROJECT_NAME}${GOEXE}'

    go 'build -o build/${GOOS}/${GOARCH}/app1a${GOEXE}  gitrepo/apps/app1/app1a'
    go 'build -o build/${GOOS}/${GOARCH}/app1b${GOEXE}  gitrepo/apps/app1/app1b'
}
```

But this does not work either. Only the last go declaration is used. 
It seems that it is not possible to define multiple go builds.

My question is, is it the intended behavior or is it a bug?

If it's the intended behavior, then which alternatives do I have?

I already tried to define custom tasks, or the new muti-project configuration, 
but it feels like it just complicates a former simple setup.

**Comment by amimas** [comment]
I think you need to organize your projects following Gradle's multi project build.

https://docs.gradle.org/current/userguide/multi_project_builds.html

There's a multi project build example that uses gogradle here

https://github.com/gogradle/examples/tree/master/multi-project

**Comment by dawi** [comment]
As I mentioned before, I already tried that but I don't like the result for multiple reasons.

* The multiproject build is less straight forward to configure.
* The sub projects are populated with {{.gogradle}} and {{vendor}} directories.

Previously (with gogradle 0.6.5) I had the following project structure:

```
git
    apps
        app1 <- git project
            .gogradle
            app1lib
            app1binA
            app1binB
            build
            vendor
```

Now with subprojects I have the following structure:

```
git
    apps
        app1 <- git project
            .gogradle
            app1lib
            app1binA
                .gogradle
                build
                vendor
            app1binB
                .gogradle
                build
                vendor
```

**Comment by amimas** [comment]
In my opinion, that's how multi project builds are supposed to be. If you apply the java or distribution plugin, the folder structure will look the same. You'll get compiled archives in a directory named build under each projects.

**Comment by dawi** [comment]
Yes, you are right. My point is not that a multi project setup is not useful. Indeed it is.

I just wanted to point out, that my relatively simple setup that worked before now is much more complicated.

An additional Point is that the IntelliJ integration is worse, because if I want to navigate from my Application Source Code to the Library Source Code, IntelliJ will send me to the vendor directory because my library is vendored as well.

Therefore my question remains: Was it a conscious decision to only allow one binary per project, because it would be nice if I could still build multiple binaries without using a multi project setup.

**Comment by blindpirate** [comment]
Well, that's kind of intentional. Previously, 

```
build {
     doLast {
          go 'cmd1'
          go 'cmd2'
     }
}
```

will simply run `go cmd1` and `go cmd2`. Since `0.7.0`, `build` task doesn't do actual work any more. It will "fork" several build tasks, e.g. `buildAmd64Windows` and `buildAmd64Linux`. In my view, a task should only do one thing at the same time, however, apparently we'd better have a convenient way to handle your case, which I'll consider improve in next release. Sorry for the inconvenience.

Multi-project doesn't look a good solution here I think.

**Comment by dawi** [comment]
@blindpirate This would be great. :)

**Comment by dawi** [comment]
@blindpirate Since 0.9 seems to be around the corner I couldn't wait to give it a try. :)

It works great!

**Thank you for your hard work on gogradle and also thank you for your hard work on the Gradle JUnit 5 integration!**

**Comment by blindpirate** [comment]
Closed by `0.9`.

@dawi Thank you, it's my pleasure.
